File: api/src/entity/Transaction.ts
import { Column, Entity, JoinTable, ManyToMany, ManyToOne, PrimaryGeneratedColumn } from 'typeorm';
import { Tag } from './Tag';
import { Upload } from './Upload';

@Entity()
export class Transaction {

    @PrimaryGeneratedColumn()
    id: number;

    @Column('date')
    date: Date;

    @Column('text')
    description: string;

    @Column('double precision', { scale: 2 })
    amount: number;

    @Column('double precision', { scale: 2 })
    balance: number;

    @Column('double precision', { scale: 2, default: 0 })
    balanceCorrection: number;

    @ManyToOne(() => Upload, upload => upload.transactions)
    upload: Upload;

    @ManyToMany(type => Tag)
    @JoinTable()
    tags: Tag[];

}

File: go-server/model/transaction.go
package model

import (
	"time"

	"gorm.io/gorm"
)

type Transaction struct {
  gorm.Model
	UploadID uint
	CategoryID uint
	Date time.Time
	Description string
	Comment *string
	Amount int64
	Balance int64
	BalanceCorrection int64 `gorm:"default:0"`
}